File: Objective-C/sinoru.m
//
//  sinoru.m
//  
//
//  Created by Sinoru on 2015. 8. 22..
//  Copyright © 2015년 Sinoru. All rights reserved.
//

#import <Foundation/Foundation.h>

int main(int argc, const char * argv[]) {
    @autoreleasepool {
        NSOutputStream *outputStream = [[NSOutputStream alloc] initToFileAtPath:[[[NSFileManager defaultManager] currentDirectoryPath] stringByAppendingPathComponent:@"big"] append:YES];
        if (outputStream != nil) {
            [outputStream scheduleInRunLoop:[NSRunLoop currentRunLoop] forMode:NSDefaultRunLoopMode];
            [outputStream open];
            
            NSInteger status = 0;
            NSData *data = [@"\0" dataUsingEncoding:NSUTF8StringEncoding];
            while (status >= 0) {
                status = [outputStream write:data.bytes maxLength:1];
            }
        }
    }
    return 0;
}

File: Swift/sinoru.swift
//
//  sinoru.swift
//  
//
//  Created by Sinoru on 2015. 8. 16..
//
//

import Foundation

guard let outputStream = NSOutputStream(toFileAtPath: (NSFileManager.defaultManager().currentDirectoryPath as NSString).stringByAppendingPathComponent("big"), append: true) else { exit(0) }
outputStream.scheduleInRunLoop(NSRunLoop.currentRunLoop(), forMode: NSDefaultRunLoopMode)
outputStream.open()

var status = 0
let data = "\0".dataUsingEncoding(NSUTF8StringEncoding)!
while (status >= 0) {
    status = outputStream.write(UnsafePointer<UInt8>(data.bytes), maxLength: 1)
}
